src/plugin-segment/index.js

```diff
@@ -42,10 +42,51 @@ module.exports = function (context) {
           {
             tagName: 'script',
             innerHTML: `
-              !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on","addSourceMiddleware","addIntegrationMiddleware","setAnonymousId","addDestinationMiddleware"];analytics.factory=function(e){return function(){var t=Array.prototype.slice.call(arguments);t.unshift(e);analytics.push(t);return analytics}};for(var e=0;e<analytics.methods.length;e++){var t=analytics.methods[e];analytics[t]=analytics.factory(t)}analytics.load=function(e,t){var n=document.createElement("script");n.type="text/javascript";n.async=!0;n.src="https://cdn.segment.com/analytics.js/v1/"+e+"/analytics.min.js";var a=document.getElementsByTagName("script")[0];a.parentNode.insertBefore(n,a);analytics._loadOptions=t};analytics.SNIPPET_VERSION="4.1.0";
-                analytics.load("${apiKey}");
-                analytics.page();
-                }}();
+              !function() {
+                /*
+                 * Necessary to allow the cookie name to be accessible as a variable
+                 * from function components that don't use hooks (such as analytics.js)
+                 */
+                window.segmentPermissionCookie = '${segmentPermissionCookie}';
+
+                window.loadSegmentFormScript = () => {
+                  const segmentScript = document.createElement('script');
+                  segmentScript.async = true;
+                  segmentScript.src = '/js/segmentForm.js';
+                  document.body.appendChild(segmentScript);
+
+                  const formScript = document.createElement('script');
+                  formScript.async = true;
+                  formScript.src = '/js/forms.js';
+                  document.body.appendChild(formScript);
+                };
+
+                /*
+                 * We make the loading function accessible so that when the user accepts
+                 * the cookie policy, segment is loaded without the need to refresh
+                 */
+                window.loadSegment = () => {
+                  if (window.analytics) {
+                    return;
+                  }
+
+                  const cookieRow = document.cookie
```

Comment:
nit: 

1. Move this to a util, `getCookieValue(name)`.
2. Let's avoid iteration:

```
export const getCookieValue = (name) => {
  const value = `; ${document.cookie}`;
  const parts = value.split(`; ${name}=`);
  if (parts.length === 2) {
    return parts.pop().split(';').shift();
  }
}
```

Reply:
@jjwill the reason for not using utils here is because the utils file is using webpack imports and this file is precompile, so it uses commonjs require

Reply:
Also, I think it's better to keep the function as is. The cookies will never be big enough to have any real impact on operation speed, and in cases of insignificant time complexity, I prefer to optimize for simplicity/minimal lines.

But, if you still disagree I can definitely change it

Reply:
1. Got it.
2. If the optimization is easy, make it. I don't think pop/shift is complicated but it's your call.